pkg/cli/subcommands/check/command.go

```diff
@@ -0,0 +1,621 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package check builds a 'check' command to be used in binaries.
+package check
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+	"go.uber.org/fx"
+	"gopkg.in/yaml.v2"
+
+	"github.com/DataDog/datadog-agent/cmd/agent/common"
+	"github.com/DataDog/datadog-agent/comp/core"
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/cli/standalone"
+	"github.com/DataDog/datadog-agent/pkg/collector"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	pkgconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/metadata/inventories"
+	"github.com/DataDog/datadog-agent/pkg/status"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+	"github.com/DataDog/datadog-agent/pkg/util/scrubber"
+)
+
+// cliParams are the command-line arguments for this subcommand
+type cliParams struct {
+	// *command.GlobalParams
+
+	// cmd is the running cobra.Command
+	cmd *cobra.Command
+
+	// args are the positional command line args
+	args []string
+
+	// subcommand-specific params
+
+	checkRate                 bool
+	checkTimes                int
+	checkPause                int
+	checkName                 string
+	checkDelay                int
+	logLevel                  string
+	formatJSON                bool
+	formatTable               bool
+	breakPoint                string
+	fullSketches              bool
+	saveFlare                 bool
+	profileMemory             bool
+	profileMemoryDir          string
+	profileMemoryFrames       string
+	profileMemoryGC           string
+	profileMemoryCombine      string
+	profileMemorySort         string
+	profileMemoryLimit        string
+	profileMemoryDiff         string
+	profileMemoryFilters      string
+	profileMemoryUnit         string
+	profileMemoryVerbose      string
+	discoveryTimeout          uint
+	discoveryRetryInterval    uint
+	discoveryMinInstances     uint
+	generateIntegrationTraces bool
+}
+
+type GlobalParams struct {
+	ConfFilePath string
+	ConfigName   string
+	LoggerName   string
+}
+
+// MakeCommand returns a `check` to be used by agent binaries.
+func MakeCommand(globalParamsGetter func() GlobalParams) *cobra.Command {
```

Comment:
If you're wondering why this takes a `func() GlobalParams` instead of just `GlobalParams`: https://github.com/DataDog/datadog-agent/blob/main/cmd/agent/command/command.go#L17-L21

`MakeCommand` can't take a `command.GlobalParams` either, since the DCA will implement its own type for global params, because it accepts less parameters than `agent`. That's also likely to be the case with the other binaries.